parsons/databases/database_connector.py

```diff
@@ -0,0 +1,108 @@
+from abc import ABC, abstractmethod
+from typing import Optional
+from parsons.etl.table import Table
+
+
+class DatabaseConnector(ABC):
+    """
+    An abstract base class that provides a uniform interface for all Parsons database connectors.
+    This class should be used in functions instead of the specific database connector classes
```

Comment:
I think readers are going to need an example of when they should use DatabaseConnector vs a specific connector class. I need an example and I understand ABCs pretty well!

Also worth thinking about whether this documentation is primarily meant for Parsons contributors or Parsons users. I assume Parsons contributors, but I could see a user reading this and getting confused.

Reply:
I have added some (possibly too) extensive documentation that hopefully makes this clearer!

Reply:
This documentation is great! Super helpful. And yes it's a lot of words but I'd rather err on the side of giving people too much information than too little.